packages/datatrak-web/src/api/mutations/useEditUser.ts

```diff
@@ -4,45 +4,36 @@
  */
 
 import { useMutation, useQueryClient } from 'react-query';
-import { Project } from '@tupaia/types';
+import { UserAccountDetails } from '../../types';
 import { put } from '../api';
-import { Entity } from '../../types';
 
-type UserPreferences = {
-  projectId?: Project['id'];
-  countryId?: Entity['id'];
-  deleteAccountRequested?: boolean;
-};
+/**
+ * Converts a string from camel case to snake case.
+ *
+ * @remarks
+ * Ignores whitespace characters, including wordspaces and newlines. Does not handle fully-
+ * uppercase acronyms/initialisms. e.g. 'HTTPRequest' -> 'h_t_t_p_request'.
+ */
+function camelToSnakeCase(camelCaseString: string): string {
+  return camelCaseString
+    ?.split(/\.?(?=[A-Z])/)
+    .join('_')
+    .toLowerCase();
+}
 
 export const useEditUser = (onSuccess?: () => void) => {
   const queryClient = useQueryClient();
 
-  return useMutation<any, Error, UserPreferences, unknown>(
-    async ({ projectId, countryId, deleteAccountRequested }: UserPreferences) => {
-      if (!projectId && !countryId && deleteAccountRequested === undefined) {
-        return;
-      }
-
-      const updates = {} as {
-        project_id?: Project['id'];
-        country_id?: Entity['id'];
-        delete_account_requested?: boolean;
-      };
-
-      if (projectId) {
-        updates.project_id = projectId;
-      }
+  return useMutation<any, Error, UserAccountDetails, unknown>(
+    async (userDetails: UserAccountDetails) => {
+      if (!userDetails) return;
 
-      if (countryId) {
-        updates.country_id = countryId;
-      }
+      const updates = Object.entries(userDetails).reduce(
+        (obj, [key, value]) => ({ ...obj, [camelToSnakeCase(key)]: value }),
+        {},
+      );
```

Comment:
I use `mobileNumber` instead of `contactNumber` because of this `camelToSnakeCase()` usage here. It works more neatly with the `mobile_number` field in the database.

However, existing variables in the codebase use `contactNumber` to refer to what is really a very similar thing. Should I update those symbol names for consistency?

Reply:
I think that is actually a good idea. It's not a problem is the label is different to the variable name or form field name but if we are using different variable names to refer to the same database field then that is confusing.

Reply:
I’m not confident this is trivial enough for me to do in the same PR; probably safer for me to do this in another branch and make sure I don’t break anything